検索機能の新規実装

mamecom opened the pull request:
検索対象は全一致の場合に限るように実装しました。

mamecom committed:
検索機能の実装（全一致のみ)

mamecom committed:
検索結果の表示にヘッダー情報を表示させるように追加しました。

hiro13255 commented on main.go:
```diff
@@ -32,11 +33,12 @@ func main() {
 // 　メニュー画面
 func Menu() int {
 	var selecter int
+	Display := true
```
実装ありがとうございます！
こちら変数であればdisplayの方が良さそうです！

mamecom replied:
ありがとうございます！
先頭が小文字になるよう修正しました。

hiro13255 commented on main.go:
```diff
@@ -46,11 +48,16 @@ func Menu() int {
 		default:
 		}
-		DisplayRecords()
+		if Display {
```
このDisplayフラグはfalseにならない挙動であってますか？
一度5を入力されたらずっとこのif文に入ってしまう気がするのですが、こちらは仕様でしょうか？

mamecom replied:
昨日助言を頂いたよう、このif文に入ってしまう挙動を修正しました。

hiro13255 commented on main.go:
```diff
@@ -281,16 +287,55 @@ func UpdateProducts(updateNo int) error {
+	DisplaySeachRecors(searchRecords)
+}
+
+func Assoc(records [][]string, str string) [][]string {
```
Assocってどういう意味でしょうか？？
もし検索実行的な意味合いだったら~Execとかの方がわかりやすくなるかも知れないです！

mamecom replied:
こちら修正できていませんでした。
Assocというのは意味がわからないため、「SearchExec」に修正しました。

hiro13255 commented on main.go:
```diff
@@ -281,16 +287,55 @@ func UpdateProducts(updateNo int) error {
+	return searchRecords
+}
+
+func DisplaySeachRecors(searchRecords [][]string) {
```
このメソッドは画面表示させるメソッドという認識であってますか？
中の処理をみると書き込みをしているようなメソッドなのかなとおもって確認したいです！

mamecom replied:
こちらの関数はDisplayRecordsメソッドと同じような処理なので、まとめていこうと思います！

mamecom committed:
検索機能の新規実装 #5　レビュー修正
　変数名「Display」を修正
　Menuメソッドのメニュー選択、実行後の条件分の修正
　CSVファイル表示メソッドが、商品全体表示と検索結果表示で2つあったのを1つに統合
　

mamecom committed:
Merge branch 'develop' into feature/search

* develop:
  ソート機能実装 #6 レビュー修正 	SortRecordsメソッドのSwitch文にdefaultパターンを追記
  ソート機能実装 	昇順、降順でCSVファイルを変更できるよう追加修正
  ソート機能実装 　バブルソートを実装
